CommonTools/PileupAlgos/src/PuppiContainer.cc

```diff
@@ -235,14 +237,17 @@ std::vector<double> const & PuppiContainer::puppiWeights() {
         //Basic Cuts
         if(pWeight                         < fPuppiWeightCut) pWeight = 0;  //==> Elminate the low Weight stuff
         if(pWeight*fPFParticles[i0].pt()   < fPuppiAlgo[pPupId].neutralPt(fNPV) && fRecoParticles[i0].id == 0 ) pWeight = 0;  //threshold cut on the neutral Pt
-        
+        if(fInvert) pWeight = 1.-pWeight;
         //std::cout << "fRecoParticles[i0].pt = " <<  fRecoParticles[i0].pt << ", fRecoParticles[i0].charge = " << fRecoParticles[i0].charge << ", fRecoParticles[i0].id = " << fRecoParticles[i0].id << ", weight = " << pWeight << std::endl;
 
         fWeights .push_back(pWeight);
         fAlphaMed.push_back(fPuppiAlgo[pPupId].median(0));
         fAlphaRMS.push_back(fPuppiAlgo[pPupId].rms(0));        
         //Now get rid of the thrown out weights for the particle collection
-        if(std::abs(pWeight) < std::numeric_limits<double>::denorm_min() ) continue;
+
+        // if( std::abs(pWeight) < std::numeric_limits<double>::denorm_min() ) continue; // this line seems not to work like it's supposed to...
```

Comment:
Would like to keep these 2 lines nin case we break the 1-to-1 ness in the future